sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaRecord.java

```diff
@@ -33,6 +34,7 @@
   private final String topic;
   private final int partition;
   private final long offset;
+  private final Headers headers;
```

Comment:
Could you explain how this would work? Marking this as ```@Nullable``` would not help circumvent the issue that ```Headers``` class would still need to be on the classpath. To verify, I made the requested changes and I receive the following runtime error:

```
Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/kafka/common/header/Headers
	at org.apache.beam.sdk.io.kafka.KafkaUnboundedSource.getOutputCoder(KafkaUnboundedSource.java:142)
```

When using kafka-clients 0.10.1.0.

Reply:
can you push your changes? `KafkaRecordCoder` constructor does not access Headers at all. It is accessed only in encode and decode (where we should handle null values for headers).

Reply:
Sure will do. See updated comment above.

Reply:
Thanks for the update. I missed later part of your comment above.

> and I package this class and use it as a dependency in another application which simply instantiates this class. Even though the downstream application did not call getHeaders, it will still throw a runtime exception.

Is that true? Can you try it with a small simple java class? I tested with a small test app and it worked ok . 
Run it with different versions of kafka, and with '-Dis.present=True|False':

````
package org.example;

import org.apache.kafka.clients.consumer.ConsumerRecord;
import org.apache.kafka.common.header.Headers;

public class TestRuntimeLoading {

  public static class A {

    Headers kafkaHeaders = null;

    Headers getHeaders(boolean isPresent) {
      if (isPresent && kafkaHeaders == null) {
        return new ConsumerRecord<>("test", 0, 0L, "key", "value").headers();
      } else {
        throw new RuntimeException("no headers");
      }
    }

  }

  public static void main(String[] args) {

    boolean isPresent = Boolean.valueOf(System.getProperty("is.present", "True"));
    Object headers = new A().getHeaders(isPresent);
    System.out.println("Headers : " + headers.toString());
  }
}
````

Reply:
Thank you for sharing this. Yes, this works as expected, but the issue comes when we initialize Headers in the if branch like so (similar to my example):

```java
if (isPresent && kafkaHeaders == null) {
        Headers headers = new RecordHeaders();
        headers.add("headerKey", "headerVal".getBytes());
        return new ConsumerRecord<>("test", 0, 0L, "key", "value").headers();
}
```
Even with is.present set to False, this throws a runtime exception.

Reply:
hmm.. may be that is one of the reasons  `ConsumerRecord.headers()` returns mutable headers. Could we just use that?